### PR TITLE
improve error for cycle in ranking expression functions

### PR DESCRIPTION
**Pull request by arnej27959**
@bratseth please review
@bjormel FYI

### CODE REVIEW

File: searchlib/src/main/java/com/yahoo/searchlib/rankingexpression/rule/ReferenceNode.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public StringBuilder toString(StringBuilder string, SerializationContext context
                 path = new ArrayDeque<>();
             String myPath = getName() + getArguments().expressions();
             if (path.contains(myPath))
-                throw new IllegalStateException("Cycle in ranking expression function: " + path);
+                throw new IllegalStateException("Cycle in ranking expression function '" + getName() + "' called from: " + path);
```
Comment (baldersheim) [inline searchlib/src/main/java/com/yahoo/searchlib/rankingexpression/rule/ReferenceNode.java:82]: Should this be IllegalArgumentException too, it is error in the input ?

Comment (bratseth) [reply searchlib/src/main/java/com/yahoo/searchlib/rankingexpression/rule/ReferenceNode.java:82]: Yes.

File: config-model/src/main/java/com/yahoo/schema/derived/DerivedConfiguration.java [hunk 2]
```diff
@@ -88,8 +88,8 @@ public DerivedConfiguration(Schema schema, DeployState deployState) {
             }
             Validation.validate(this, schema);
         }
-        catch (IllegalArgumentException e) {
-            throw new IllegalArgumentException("Invalid " + schema, e);
+        catch (IllegalArgumentException|IllegalStateException e) {
+            throw new IllegalArgumentException("Invalid " + schema + " -> " + e.getMessage(), e);
```
Comment (bratseth) [inline config-model/src/main/java/com/yahoo/schema/derived/DerivedConfiguration.java:92]:
```suggestion
            throw new IllegalArgumentException("Invalid " + schema, e);
```

**Review by bratseth — COMMENTED** [review]
Otherwise LGTM

**Pull request merged**